### DISCUSSION TITLE
Named HTTP-parameters in routes for API Gateway?

### DISCUSSION THREAD

**Discussion by Powermind** [brefphp/bref#1198] [Support] [answered]
Hi guys!

Sorry for this very basic questions, but Google wasn't very helpful and I am very new to Bref/Lambda/Api gateway.

I have a (very) small PHP web app on AWS lambda using Bref. Application works perfectly (generates a PDF from some input parameters) but I am unsure of how to handle nicer-looking URLs.

If I make a normal GET request to my app like this:
`https://<site>/?code=ABC&value=200&date=2023-05-05`

everything works perfectly, but I would like to have it in the format:

`https://<site>/ABC/200/2023-05-05`

I tried this in my serverless.yml:

```
        ...
        events:
            -   httpApi: 'GET /{value}/{code}/{date}'
        ...
```

But that doesn't seem to work. 
Edit: What I mean by this is that the PHP does not receive the parameters (they are not in the $_GET or $_REQUEST-variables).

Can anyone point me in the right direction? Do I need to do some custom work in API Gateway or is there some simple way to get this to work using just serverless.yml?

Thankful for any help!

**Comment by mnapoli** [comment, marked as answer]
That looks about right.

> But that doesn't seem to work.

We cannot help without more details 😄

**Reply by Powermind** [reply]
Apologies. What happens is that the app takes the default values for the parameters, as if they weren't sent at all :(
Logs show nothing weird, for some reason the GET-request does not contain any parameters (according to the PHP at least). Original question edited to reflect this.

**Reply by mnapoli** [reply]
Oh ok, it's documented here then: https://bref.sh/docs/function/handlers.html#api-gateway-http-events

<img width="763" alt="image" src="https://user-images.githubusercontent.com/720328/167130303-9cf4e425-e013-4d1b-b958-6ecdb66f3728.png">

But that works if you define API Gateway HTTP handlers as classes.

You mention you use `$_GET`, so I'm guessing you use the FPM layer. In that case I'm not even sure that's possible 🤔  That's because it wasn't really planned for that kind of use case.

Basically I would recommend this:

- either use FPM (PHP "as usual", like on any server), and in that case you use a router library or framework (i.e. your code does the routing based on the actual URL
- or either use API Gateway routing + a typed API Gateway handler, but that's usually a bit more advanced/complex

**Reply by Powermind** [reply]
Thank you, then I know where to go from here!
